### PR TITLE
Move system-probe to cobra commands

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

It transitions system-probe to using `cobra` commands/subcommands versus plain flag usage. 

### Motivation

This is in preparation for adding subcommands to change runtime configuration and runtime debugging.

### Additional Notes

The way to perform a couple actions has changed:
1. Running is now `system-probe run` (mirrors core agent)
2. Version information is accessed via `system-probe version` now.

### Describe your test plan

system-probe should start and accept the same command line parameters as before (`-config` and `-pid`).

### CODE REVIEW

**Review by heyronhay — APPROVED** [review]

File: cmd/system-probe/app/run.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
+package app
```
Comment (hmahmood) [inline cmd/system-probe/app/run.go:1]: How much of this file is existing code?

**Review by hmahmood — COMMENTED** [review]
It would help with the review if there were more details of the changes. The diffs are large, and I am having a hard time telling what has really changed other than the command.

File: cmd/system-probe/app/run.go (hunk 1 above)
Comment (brycekahle) [reply cmd/system-probe/app/run.go:1]: It is mostly moved from the previous `main.go`, but they are small differences to handle the different control flow.

**Review by hush-hush — APPROVED** [review]

**Review by albertvaka — APPROVED** [review]
Please add a release note mentioning the system-probe now takes a `run` command. It's quite unlikely but some users might have custom startup scripts that this change could break.

**Review by shang-wang — APPROVED** [review]
👍 for processes

**Comment by brycekahle** [comment]
@albertvaka @hush-hush I ended up being able to make `run` the default subcommand if none is specified. I've removed the changelog warning since it will be backwards-compatible now.

**Review by vboulineau — APPROVED** [review]

File: Dockerfiles/manifests/all-containers/daemonset.yaml [hunk 2]
```diff
@@ -261,7 +261,7 @@ spec:
             - NET_RAW
             - IPC_LOCK
           privileged: false
-        command: ["/opt/datadog-agent/embedded/bin/system-probe", "--config=/etc/datadog-agent/system-probe.yaml"]
+        command: ["/opt/datadog-agent/embedded/bin/system-probe", "run", "--config=/etc/datadog-agent/system-probe.yaml"]
```
Comment (vboulineau) [inline Dockerfiles/manifests/all-containers/daemonset.yaml:264]: You should not change the `manifests` just yet

**Review by vboulineau — CHANGES_REQUESTED** [review]
Manifests change should be done later

File: Dockerfiles/manifests/all-containers/daemonset.yaml (hunk 2 above)
Comment (brycekahle) [reply Dockerfiles/manifests/all-containers/daemonset.yaml]: Changes undone.

**Review by vboulineau — APPROVED** [review]

**Pull request merged**